python/ray/_private/function_manager.py

```diff
@@ -127,6 +131,18 @@ def load_function_or_class_from_local(self, module_name,
         except Exception:
             return None
 
+    def export_key(self, key):
+        """Export a key so it can be imported by other workers"""
+        pos = self._worker.import_thread.num_imported
+        while True:
+            pos += 1
+            holder = make_export_key(pos)
+            if self._worker.gcs_client.internal_kv_put(
```

Comment:
Is this guaranteed to be atomic set if not exists?

Reply:
It's running in a single thread (io pool) in GCS and we don't have replicas, so I think it's guaranteed.
Later the transaction guarantee should be taken care of when we are introducing other storage.